### DISCUSSION TITLE
救命！！umi3 根据文件路径自动生成路由，该怎么实现路由懒加载呀

### DISCUSSION THREAD

**Discussion by penny006** [umijs/umi#11961] [Q&A]
config.js里已开启dynamicImport
路由不是一个个页面配置的，而是根据路径把pages目录下的文件自动生成路由，这种方式该怎么实现懒加载呀？尝试半天都不成功，有没有大佬遇到过呢？
const findComponents = (files: __WebpackModuleApi.RequireContext) => {
  const modules = {};
  files.keys().forEach((key) => {
    // 删除路径开头的./ 以及结尾的 /index；
    const str = key.replace(/(\.\/|\.tsx)/g, '').replace('/index', '');
    // modules[str] = files(key).default;
    modules[str] = dynamic({
      loader: async function () {
        const { default: Comp } = await import(
          /* webpackChunkName: "oprecord" */ `../../pages/${str}/`
        );
        return Comp;
      },
    });
    // modules[str] =React.createElement(require(`../../pages/${str}/`).default)
    // modules[str] =require(`../../pages/${str}/`).default
  });
  return modules;
};

**Comment by fz6m** [comment]
umi 3 不维护了，建议升级到 umi 4
